src/Servers/HttpSys/src/ClientCertificateMode.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Server.HttpSys
+{
+    /// <summary>
+    /// Describes the client certificate negotiation methods a HTTPS connection.
+    /// </summary>
+    public enum ClientCertificateMode
+    {
+        /// <summary>
+        /// A client certificate will not be populated on the request.
+        /// </summary>
+        NoCertificate = 0,
+
+        /// <summary>
+        /// A client certificate will be populated if already present at the start of a request.
+        /// </summary>
+        AllowCertificate,
+
+        /// <summary>
+        /// The TLS session can be renegotiated to request a client certificate.
+        /// </summary>
+        DynamicCertificate
```

Comment:
Will most people know DynamicCertificate means? Maybe it would be better to name this RenegotiateCertificate given that renegotiation happens even when the cert is sent initially. I think that's what I remember you saying at least.

Reply:
That's some good naming bikeshedding :). I do prefer the name `RenegotiateCertificate`.

Reply:
> given that renegotiation happens even when the cert is sent initially. I think that's what I remember you saying at least.

I got into the wireshark traces and confirmed that the renegotiation only happens if a certificate was _not_ requested/provided at the start of a connection.